tests/helpers/test_selector.py

```diff
@@ -359,7 +359,26 @@ def test_text_selector_schema(schema, valid_selections, invalid_selections):
                 ]
             },
             ("red", "green"),
-            ("cat", 0, None),
+            ("cat", 0, None, ["red"]),
+        ),
+        (
+            {"options": ["red", "green", "blue"], "multiple": True},
+            (["red"], ["green", "blue"]),
+            ("cat", 0, None, "red"),
+        ),
+        (
+            {
+                "options": ["red", "green", "blue"],
+                "multiple": True,
+                "custom_value": True,
+            },
+            (["red"], ["green", "blue"], ["red", "cat"]),
+            ("cat", 0, None, "red"),
+        ),
+        (
+            {"options": ["red", "green", "blue"], "custom_value": True},
+            ("red", "green", "blue", "cat"),
+            (0, None, ["red"]),
         ),
```

Comment:
Let's add tests showing an empty options list is allowed:
```suggestion
        (
            {"options": ["red", "green", "blue"], "custom_value": True},
            ("red", "green", "blue", "cat"),
            (0, None, ["red"]),
        ),
        (
            {"options": [], "custom_value": True},
            ("red", "cat"),
            (0, None, ["red"]),
        ),
        (
            {"options": [], "custom_value": True, "multiple": True},
            (["red"], ["green", "blue"], []),
            (0, None, ["red"]),
        ),
```